internal/characters/arlecchino/arlecchino.go

```diff
@@ -0,0 +1,130 @@
+package arlecchino
+
+import (
+	"fmt"
+
+	tmpl "github.com/genshinsim/gcsim/internal/template/character"
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+	"github.com/genshinsim/gcsim/pkg/core/info"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/model"
+)
+
+func init() {
+	core.RegisterCharFunc(keys.Arlecchino, NewChar)
+}
+
+type char struct {
+	*tmpl.Character
+	skillDebt             float64
+	skillDebtMax          float64
+	initialDirectiveLevel int
+	chargeEarlyCancelled  bool
+}
+
+func NewChar(s *core.Core, w *character.CharWrapper, _ info.CharacterProfile) error {
+	c := char{}
+	c.Character = tmpl.NewWithWrapper(s, w)
+
+	c.EnergyMax = base.SkillDetails.BurstEnergyCost
+	c.NormalHitNum = normalHitNum
+	c.NormalCon = 3
+	c.BurstCon = 5
+
+	w.Character = &c
+
+	c.chargeEarlyCancelled = false
+
+	return nil
+}
+
+func (c *char) Init() error {
+	c.naBuff()
+	c.passive()
+	c.a1OnKill()
+
+	c.a4()
+
+	c.c2()
+	c.c6()
+	return nil
+}
+
+func (c *char) NextQueueItemIsValid(k keys.Char, a action.Action, p map[string]int) error {
```

Comment:
i think missing a check for swap here?

also more of a sanity check (unless next action is swap) but should consider checking if `k` is current char key

Reply:
Why is a special check for swap needed here?